src/arweave/composite-client.ts

```diff
@@ -441,9 +441,9 @@ export class ArweaveCompositeClient
 
             // Prefetch transactions
             if (prefetchTxs) {
-              block.txs.forEach(async (txId: string) => {
-                this.prefetchTx({ txId });
-              });
+              for (const txId of block.txs) {
+                await this.prefetchTx({ txId });
```

Comment:
The lack of `await` here was intentional. We want to "fire and forget" all the TX prefetches when a block is retrieved. The promises get awaited later then `getTx` is called.

Reply:
I see, I felt it wasn't intentional and was an extra layer of flooding. Yeh I'll remove the await and add comment about this.

Reply:
done